### ISSUE TITLE
tslint-cli broken on current master

### ISSUE THREAD

**Issue by adidahiya** [palantir/tslint#424]
trying to run `./bin/tslint-cli.js` on a file results in an error. this is due to incorrect configuration of the grunt `bin` task.

**Comment by leeavital** [comment]
I have a possible fix at https://github.com/leeavital/tslint/tree/cli-fix. The problem is that RuleWalker (which extends SyntaxWalker) is being concatenated before SyntaxWalker, so it extending a class that doesn't exist. Making sure RuleWalker explicitly reference SyntaxWalker with a <reference /> fixes the problem, although it seems like grunt-ts should take care of this.

**Comment by adidahiya** [comment]
We used to have reference paths all over this project, but I removed them recently when adding `tsconfig.json` project files in a few places. Ideally, grunt-ts should use these project files for compilation, but for now we have to duplicate the build configuration into `Gruntfile.js`. Given that, we should be able to change the config for the `ts:bin` task to make it look like this (to match `ts:core`):

```js
            bin: {
                src: [
                    "typings/*.d.ts",
                    "src/language/walker/syntaxWalker.ts",
                    "src/language/**/*.ts",
                    "src/*.ts",
                ],
                out: "bin/tslint-cli.js"
            }
```

**Comment by leeavital** [comment]
That did it! I also had for force tslint-cli.js to load last.

To prevent this from happening again I've added a grunt task that makes sure tslint exits with the correct codes in  #428.

**Comment by adidahiya** [comment]
great! thanks for adding the automated test for it
